slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -431,6 +431,75 @@ void DetectorImpl::setGapPixelsinCallback(const bool enable) {
     shm()->gapPixels = enable;
 }
 
+int DetectorImpl::getTransmissionDelay() const {
+    bool eiger = false;
+    switch (shm()->detType) {
+    case JUNGFRAU:
+    case MYTHEN3:
+        break;
+    case EIGER:
+        eiger = true;
+        break;
+    default:
+        throw RuntimeError(
+            "Transmission delay is not implemented for the this detector.");
+    }
+    if (!eiger && size() <= 1) {
+        throw RuntimeError(
+            "Cannot get intermodule transmission delays with just one module");
+    }
+    int step = 0;
+    if (eiger) {
+        // between left and right
+        step = modules[0]->getTransmissionDelayRight();
+    } else {
+        // between first and second
+        step = modules[1]->getTransmissionDelayFrame();
+    }
+    for (int i = 0; i != size(); ++i) {
+        if (eiger) {
+            if ((modules[i]->getTransmissionDelayLeft() != (3 * i * step)) ||
+                (modules[i]->getTransmissionDelayRight() !=
+                 ((3 * i + 1) * step)) ||
+                (modules[i]->getTransmissionDelayFrame() !=
+                 ((3 * i + 2) * step))) {
+                throw RuntimeError("Could not get the transmission delay. "
+                                   "Unequal delays between modules.");
+            }
+        } else {
+            if (modules[i]->getTransmissionDelayFrame() != (i * step)) {
+                throw RuntimeError("Could not get the transmission delay. "
+                                   "Unequal delays between modules.");
+            }
+        }
+    }
+    return step;
+}
+
+void DetectorImpl::setTransmissionDelay(int step) {
+    bool eiger = false;
+    switch (shm()->detType) {
+    case JUNGFRAU:
+    case MYTHEN3:
+        break;
+    case EIGER:
+        eiger = true;
+        break;
+    default:
+        throw RuntimeError(
+            "Transmission delay is not implemented for the this detector.");
+    }
+    for (int i = 0; i != size(); ++i) {
+        if (eiger) {
+            modules[i]->setTransmissionDelayLeft(3 * i * step);
+            modules[i]->setTransmissionDelayRight((3 * i + 1) * step);
+            modules[i]->setTransmissionDelayFrame((3 * i + 2) * step);
```

Comment:
In our old scripts we had a fixed txdelay_frame. I think we need it for multi frame acq but maybe clarify with Martin?

```bash
for i in `seq 0 $((NUM_MOD-1))`; do
	sls_detector_put $i:txndelay_left  $((i*2*DELAY_DIFF))
	sls_detector_put $i:txndelay_right $(((i*2+1)*DELAY_DIFF))
done
       sls_detector_put txndelay_frame  $((NUM_MOD*2*DELAY_DIFF))


```

Reply:
done